.github/workflows/test-go-task.yml

```diff
@@ -89,6 +89,9 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           version: 3.x
 
+      - name: Build arduino-cli
+        run: task go:build
+
```

Comment:
Move this to the `go:test` task, as is done in the `go:test-integration` task:

https://github.com/arduino/arduino-cli/blob/c6cfe21390025b8bc8468ffd1bce2edc0ca8b17b/Taskfile.yml#L107-L110

Otherwise people running the task locally will have a bad time when the test either fails due to the missing executable, or far worse when they get incorrect test results because they don't realize it is using whichever random executable happened to be sitting around from the last time they did a build (I had that happen to me back before we added the build to the integration test task).